wgpu/src/backend/wgpu_core.rs

```diff
@@ -645,13 +663,14 @@ impl ErrorSinkRaw {
                 if scope.error.is_none() {
                     scope.error = Some(err);
                 }
+                None
             }
             None => {
-                if let Some(custom_handler) = self.uncaptured_handler.as_ref() {
-                    (custom_handler)(err);
+                if let Some(custom_handler) = self.uncaptured_handler.clone() {
```

Comment:
There's an idiom here: when a clone is targeting an `Arc`, write it out as `Arc::clone(&self.uncaptured_handler)`, so it's obvious to the reader why it's cheap, and so you'll get a compile-time error if it's not doing what you expect.

```suggestion
                if let Some(custom_handler) = Arc::clone(&self.uncaptured_handler) {
```

Reply:
If this is a strong position of yours, maybe we should discuss enabling [`clippy::clone_on_ref_ptr`](https://rust-lang.github.io/rust-clippy/stable/index.html#clone_on_ref_ptr)?

Reply:
The suggestion broke the build, because the original code is actually cloning `Option<Arc>`, not `Arc`. We could just fix that by setting the right type:

```rust
if let Some(custom_handler) = Option::<Arc>::clone(&self.uncaptured_handler) {
```

or we could defer the clone entirely:

```rust
if let Some(custom_handler) = &self.uncaptured_handler {
    let custom_handler = Arc::clone(&custom_handler);
```

These are the same at run time, but I am inclined to the second on the principle of “when in doubt, write clear, boring code”. What do you think?

Reply:
Oh, yeah, the second is clearly better.

Reply:
> If this is a strong position of yours, maybe we should discuss enabling [`clippy::clone_on_ref_ptr`](https://rust-lang.github.io/rust-clippy/stable/index.html#clone_on_ref_ptr)?

It does seem clearer to me, but I wouldn't say it's a strong opinion. I think we should consider it.

Reply:
I put this on the maintainers' meeting agenda for tomorrow.

Reply:
This question doesn’t block this PR, right?

Reply:
@kpreid: Changing `clippy` lints is definitely outta scope for this PR, don't worry about it. 🙂